### PR TITLE
features/CIV-4746-expert report details

### PR DESCRIPTION
**Pull request by laperla11**
**Before creating a pull request make sure that:**

- [x] commit messages are meaningful and follow good commit message guidelines
- [ ] README and other documentation has been updated / added (if needed)
- [x] tests have been updated / new tests has been added (if needed)

Please remove this line and everything above and fill the following sections:

### JIRA link ###
https://tools.hmcts.net/jira/browse/CIV-4746

### Change description ###
DQ - Small Claims - Have you already got a report written by an expert
- 

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[x] No
```

### CODE REVIEW

File: src/main/common/models/directionsQuestionnaire/experts/experts.ts [hunk 1]
```diff
@@ -0,0 +1,5 @@
+import {ExpertReportDetails} from './expertReportDetails/expertReportDetails';
+
+export class Experts {
+  expertReportDetails: ExpertReportDetails;
```
Comment (commitBlob) [inline src/main/common/models/directionsQuestionnaire/experts/experts.ts:4]: is experts class going to have more properties?

File: src/main/modules/i18n/locales/en.json [hunk 2]
```diff
@@ -1436,6 +1446,7 @@
     "CLAIM_DETAILS": "Claim details",
     "CLAIM_NUMBER_REQUIRED": "You need to enter the claim number",
     "CLAIM_VALID_AMOUNT": "Enter a valid amount",
+    "CORRECT_DATE_NOT_IN_FUTURE": "Correct the date. You can’t use a future date.",
```
Comment (commitBlob) [inline src/main/modules/i18n/locales/en.json:1449]: seems like you didnt add error translation keys into Welsh

File: src/main/routes/features/directionsQuestionnaire/expertReportDetailsController.ts [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+  getExpertReportDetailsForm,
+  saveExpertReportDetails,
+} from '../../../services/features/directionsQuestionnaire/expertReportDetailsService';
+import {constructResponseUrlWithIdParams} from '../../../common/utils/urlFormatter';
+import {YesNo} from '../../../common/form/models/yesNo';
+
+const expertReportDetailsController = express.Router();
+
+function renderView(form: GenericForm<ExpertReportDetails>, res: express.Response): void {
+  const expertReportDetailsForm = Object.assign(form);
```
Comment (commitBlob) [inline src/main/routes/features/directionsQuestionnaire/expertReportDetailsController.ts:16]: you don't need to do this anymore, just do `{form, today: new Date()}` in the render
and do the change in the nunjucks view

File: src/main/views/features/directionsQuestionnaire/expert-report-details.njk [hunk 4]
```diff
@@ -0,0 +1,74 @@
...
+{% from "govuk/components/button/macro.njk" import govukButton %}
+{% from "govuk/components/inset-text/macro.njk" import govukInsetText %}
+{% from "govuk/components/textarea/macro.njk" import govukTextarea %}
+{% from "govuk/components/date-input/macro.njk" import govukDateInput -%}
+{% from "govuk/components/fieldset/macro.njk" import govukFieldset -%}
+{% from "govuk/components/input/macro.njk" import govukInput -%}
+{% from "../../macro/csrf.njk" import csrfProtection %}
+{% from "../../macro/contact-us-for-help.njk" import contactUsForHelp %}
+{% from "../../macro/page-title.njk" import setPageTitle %}
+{% from "../../macro/yesNoForm.njk" import yesNoRadioButton %}
```
Comment (commitBlob) [inline src/main/views/features/directionsQuestionnaire/expert-report-details.njk:11]: replace `yesNoForm` with `yesNoGenericForm` and then you don't need object.assign in the controller

File: src/main/views/features/directionsQuestionnaire/expert-report-details.njk [hunk 5]
```diff
@@ -0,0 +1,74 @@
+{% extends "claim-details-tpl.njk" %}
+{% from "govuk/components/button/macro.njk" import govukButton %}
```
Comment (commitBlob) [inline src/main/views/features/directionsQuestionnaire/expert-report-details.njk:2]: do you need all those imports?

File: src/main/common/models/directionsQuestionnaire/experts/experts.ts (hunk 1 above)
Comment (laperla11) [reply src/main/common/models/directionsQuestionnaire/experts/experts.ts:4]: yes, there are more screens need this class, we also need to port some properties from `directionQuestionnaire` to `experts`.

File: src/main/modules/i18n/locales/en.json (hunk 2 above)
Comment (laperla11) [reply src/main/modules/i18n/locales/en.json:1497]: done

File: src/main/routes/features/directionsQuestionnaire/expertReportDetailsController.ts (hunk 3 above)
Comment (laperla11) [reply src/main/routes/features/directionsQuestionnaire/expertReportDetailsController.ts:16]: changed to `yesNoGenericForm`

File: src/main/views/features/directionsQuestionnaire/expert-report-details.njk (hunk 5 above)
Comment (laperla11) [reply src/main/views/features/directionsQuestionnaire/expert-report-details.njk:2]: removed the unused imports

File: src/main/views/features/directionsQuestionnaire/expert-report-details.njk (hunk 4 above)
Comment (laperla11) [reply src/main/views/features/directionsQuestionnaire/expert-report-details.njk]: changed to yesNoGenericForm

**Review by laarmada — APPROVED** [review]

**Pull request merged**